bindings/java/src/org/sleuthkit/datamodel/CommunicationsManager.java

```diff
@@ -381,17 +383,50 @@ public void addRelationships(AccountFileInstance sender, List<AccountFileInstanc
 						+ "Recipient source ID" + recipient.getDataSourceObjectID() + " != relationship source ID" + sourceArtifact.getDataSourceObjectID());
 			}
 		}
+		
+		// Set up the query for the prepared statement
+		String query = "INTO account_relationships (account1_id, account2_id, relationship_source_obj_id, date_time, relationship_type, data_source_obj_id  ) "
+						+ "VALUES (?,?,?,?,?,?)";
+		switch (db.getDatabaseType()) {
+			case POSTGRESQL:
+				query = "INSERT " + query + " ON CONFLICT DO NOTHING";
+				break;
+			case SQLITE:
+				query = "INSERT OR IGNORE " + query;
+				break;
+			default:
+				throw new TskCoreException("Unknown DB Type: " + db.getDatabaseType().name());
+		}		
+		
+		CaseDbTransaction trans = db.beginTransaction();	
+		try {
+			SleuthkitCase.CaseDbConnection connection = trans.getConnection();
+			PreparedStatement preparedStatement = connection.getPreparedStatement(query, Statement.NO_GENERATED_KEYS);
+			
```

Comment:
I believe that both CaseDatabase and PreparedStatement are AutoCloseable so the Codacy issue regarding closing resources could be resolved with a try-with-resources.

Reply:
I can put the CaseDbConnection part in a try with resources, but the prepared statement really shouldn't be. The transaction holds onto those for future use and closes them itself when it is closed (yes that's at the same time in our case, but we still shouldn't be manually closing it).

Reply:
Actually the CaseDbTransaction closes the connection itself, so it seems pointless to put that in a try-with-resources. It's part of the CaseDbTransaction object, along with its prepared statements.